datamap/mv/core/src/main/scala/org/apache/carbondata/mv/datamap/MVHelper.scala

```diff
@@ -596,12 +596,20 @@ object MVHelper {
         case g: GroupBy =>
           MVHelper.updateDataMap(g, rewrite)
       }
-      // TODO Find a better way to set the rewritten flag, it may fail in some conditions.
-      val mapping =
-        rewrittenPlan.collect { case m: ModularPlan => m } zip
-        updatedDataMapTablePlan.collect { case m: ModularPlan => m }
-      mapping.foreach(f => if (f._1.rewritten) f._2.setRewritten())
 
+      // Use coarse equal to set the rewritten flag
+      // We may change the ModularPlan relation
```

Comment:
comment is not very clear, can you please rewrite this?

Reply:
@akashrn5  I suggest you compare the old code with new code and you will get it